cscs-checks/microbenchmarks/gpu/memory_bandwidth/memory_bandwidth.py

```diff
@@ -6,163 +6,122 @@
 import reframe.utility.sanity as sn
 import reframe as rfm
 
+import library.microbenchmarks.gpu.memory_bandwidth as mb
+import cscslib.microbenchmarks.gpu.hooks as hooks
 
-@rfm.simple_test
-class GpuBandwidthCheck(rfm.RegressionTest):
-    def __init__(self):
-        self.valid_systems = ['daint:gpu', 'dom:gpu',
-                              'arolla:cn', 'tsa:cn',
-                              'ault:amdv100', 'ault:intelv100',
-                              'ault:amda100', 'ault:amdvega']
-        self.valid_prog_environs = ['PrgEnv-gnu']
+
+class PrgEnvMixin(rfm.RegressionMixin):
```

Comment:
The name of this mixing is not very generic. I wonder if it makes sense to leave it like this, or set more specific names and have the mixings be reused by other library classes too.
Because the name `PrgEnvMixin` tells me nothing and this one is very, very specific for Tsa and Arolla. But I wonder how this particular one will evolve!

What do you think? Do you think that this mixing will have additional functions in the future that can be generically used by other classes or that this mixing is meant to be used just to fix the Tsa and Arolla programming environment definitions?

Reply:
I have thought of having site-wide hooks that could be used for all the tests that we run at CSCS. The issue with this is that it'd be rather difficult to have a Mixin that all tests are happy with. We might need to structure the Mixin classes into their own hierarchies as well.

Regarding the naming of this mixin, so far this is the only test that uses it, so I'm not sure it makes much difference to change it now.

Reply:
I would still change the name to make it more meaningful. This also helps in understanding what the Check is depending on. It is much easier to read/reason about a test that inherits from `TsaArollaNoMPIPrgEnvMixin`, then from just `PrgEnvMixin`. Even though the name that I propose is not very good! 🤔

Reply:
I agree. But I think it's not such a big deal here because everything is on this same file, and also this is our final specialisation of the test. In any case, I don't mind changing this.

Reply:
Changed this one to `BaseCSCS`